File: src/main/java/ru/pearx/pmdumper/dumpers/DumperSounds.java
package ru.pearx.pmdumper.dumpers;

import net.minecraft.util.ResourceLocation;
import net.minecraft.util.SoundEvent;
import net.minecraftforge.fml.common.registry.ForgeRegistries;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import ru.pearx.pmdumper.dumpers.base.IDumper;
import ru.pearx.pmdumper.dumpers.base.PMDData;

import java.util.*;

/**
 * Created by mrAppleXZ on 12.06.17 12:02.
 */
@SideOnly(Side.CLIENT)
public class DumperSounds implements IDumper
{
    @Override
    public String getName()
    {
        return "sounds";
    }

    @Override
    public PMDData getData()
    {
        List<List<String>> lst = new ArrayList<>();
        Map<String, Integer> counts = new HashMap<>();
        lst.add(Arrays.asList("ID", "Name"));
        for(Map.Entry<ResourceLocation, SoundEvent> entr : ForgeRegistries.SOUND_EVENTS.getEntries())
        {
            ResourceLocation loc = entr.getKey();
            SoundEvent ev = entr.getValue();
            lst.add(Arrays.asList(loc.toString(), ev.getSoundName().toString()));
            PMDData.plusCounts(counts, loc.getResourceDomain(), 1);
        }
        return new PMDData(lst, counts);
    }
}

File: src/main/kotlin/ru/pearx/pmdumper/dumper/standard/DumperSounds.kt
@file:JvmMultifileClass
@file:JvmName("StandardDumpers")

package ru.pearx.pmdumper.dumper.standard

import net.minecraft.util.ResourceLocation
import net.minecraftforge.fml.common.registry.ForgeRegistries
import ru.pearx.pmdumper.ID
import ru.pearx.pmdumper.dumper.dumper
import ru.pearx.pmdumper.utils.tryDump

val DumperSounds = dumper {
    registryName = ResourceLocation(ID, "sounds")
    header = listOf("ID")
    amounts {
        for (sound in ForgeRegistries.SOUND_EVENTS)
            this += registryName
    }
    iterator {
        for (sound in ForgeRegistries.SOUND_EVENTS) {
            tryDump(ArrayList(header.size)) {
                with(sound) {
                    add(registryName.toString())
                }
            }
        }
    }
}